### PR TITLE
refactor(express): extract class into functions

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

Remove "class", export 2 middlewares: handleAuthRoutes and withLogto

```
app.use(handleAuthRoutes(config));
app.use(withLogto(config));
```

## Testing

UTs and express sample.

### CODE REVIEW

File: packages/express-sample/src/app.ts [hunk 1]
```diff
@@ -1,10 +1,16 @@
 import http from 'http';
 
+import { handleAuthRoutes, withLogto } from '@logto/express';
 import cookieParser from 'cookie-parser';
 import express, { Request, Response, NextFunction } from 'express';
 import session from 'express-session';
 
-import { logtoClient } from './logto';
+const config = {
```
Comment (charIeszhao) [inline packages/express-sample/src/app.ts:8]: nit: `const config: LogtoConfig = { ... }`

File: packages/express/src/index.test.ts [hunk 2]
```diff
@@ -56,56 +56,36 @@ describe('Express', () => {
...
-    it('should call client.handleSignInCallback and redirect to home', async () => {
-      const client = new LogtoClient(configs);
-      await testMiddleware({
-        middleware: client.handleSignInCallback(),
-        url: '/api/logto/sign-in-callback',
-        test: async ({ response }) => {
-          expect(response.redirect).toHaveBeenCalledWith(configs.baseUrl);
-        },
+    describe('handleSignInCallback', () => {
+      it('should call client.handleSignInCallback and redirect to home', async () => {
```
Comment (charIeszhao) [inline packages/express/src/index.test.ts:69]: nit: redirect home or redirect to home page

File: packages/express/src/index.ts [hunk 3]
```diff
@@ -15,92 +15,78 @@ export type Middleware = (
...
-      const nodeClient = this.createNodeClient(request, response);
+const createNodeClient = (
+  request: IncomingMessage,
+  response: Response,
+  config: LogtoExpressConfig
+) => {
+  // We assume that `session` is configured in the express app, but need to check it there.
+  // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+  if (!request.session) {
+    throw new LogtoExpressError('session_not_configured');
```
Comment (charIeszhao) [inline packages/express/src/index.ts:26]: nit: maybe `session not found`?

Comment (charIeszhao) [reply packages/express/src/index.ts:26]: Also why do we need to maintain a new error class `LogtoExpressError`? Can we re-use `LogtoClientError`?

**Review by charIeszhao — APPROVED** [review]

File: packages/express/src/index.ts (hunk 3 above)
Comment (wangsijie) [reply packages/express/src/index.ts:26]: 1. `session not found` sounds like the user has no session, `not configured` is more accurate.
2. `LogtoClientError` don't have this error key.

Comment (charIeszhao) [reply packages/express/src/index.ts:27]: Well, OK

**Pull request merged**